### DISCUSSION TITLE
Distinguish between client- and server-side call of `getServerSideProps`

### DISCUSSION THREAD

**Discussion by stigkj** [vercel/next.js#12240] [Help] [answered]
Is it possible to determine whether a call to `getServerSideProps` comes from the initial load of the page or from a page navigation?

We are caching client-side some parts of the data returned from `getServerSideProps` as it is used on all pages. Some data is different between pages, so on a page navigation we would like to return only the data specific to each page.

**Comment by claus** [comment, marked as answer]
This is more or less how Next does it internally:

`const isClientSide = url.parse(req.url).pathname.match(/_next\/data/);`
